commands/client_daemon_test.go

```diff
@@ -230,3 +237,57 @@ func TestPieceRejectionInProposeStorageDeal(t *testing.T) {
 
 	assert.Contains(proposeDealErrors, "piece is 3000 bytes but sector size is 1016 bytes")
 }
+
+func TestSelfDialStorageGoodError(t *testing.T) {
+	tf.IntegrationTest(t)
+
+	assert := assert.New(t)
+	require := require.New(t)
+
+	ctx := context.Background()
+
+	ctx, env := fastesting.NewTestEnvironment(ctx, t, fast.EnvironmentOpts{})
+	// Teardown after test ends.
+	defer func() {
+		err := env.Teardown(ctx)
+		require.NoError(err)
+	}()
+
+	// Start mining.
+	miningNode := env.RequireNewNodeWithFunds(1000)
+
+	// This is what mining start should do, but FAST uses mining once
+	// for some very helpful builtins and because of issue 2579 we need to
+	// mine once in a loop instead of calling start.  Once #2579 is fixed
+	// this can be replaced with start.
+	minerCreateDoneCh := make(chan struct{})
```

Comment:
Non-blocking, but is there a reason you don't want to use a context here?

Reply:
No I would also be happy to use a new context and cancel function here.